funderfinder/utils/list_gsoc.py

```diff
@@ -0,0 +1,263 @@
+    """
+    listing_page = requests.get("https://www.google-melange.com/archive/gsoc")
+    soup = bs4.BeautifulSoup(listing_page.text, features="html.parser")
+    year_link_containers = get_early_archive_listing_links(soup)
+    for container in year_link_containers:
+        year_link = extract_listing_link(container)
+        if not year_link:
+            continue
+        print(f"Getting projects for {year_link}")
+        year_projects = get_early_archive_year_projects(year_link)
+        for project in year_projects:
+            yield project
+
+
+def get_modern_archive_project(year: int, slug: str) -> dict:
+    """
+    Retrieves project metadata for a project and year after 2016.
+    :param year: Year we want to retreive project metadata from
+    :param slug: The GSOC project slug, retrieved from their API
+    :return: Dict of project metadata
+    """
+    time.sleep(SCRAPE_DELAY)
+    project_url = f"https://summerofcode.withgoogle.com/api/archive/programs/{year}/organizations/{slug}/"
+    meta = requests.get(project_url).json()
+    repos = []
+    repos.extend(get_link_matches(meta["description_html"]))
+    repos.extend(get_link_matches(meta["ideas_list_url"]))
+    for student_project in meta["projects"]:
+        repos.extend(get_link_matches(student_project["abstract_html"]))
+        repos.extend(get_link_matches(student_project["project_code_url"]))
+    return {
+        "name": meta["name"],
+        "link": project_url,
+        "repos": list(set(repos)),
+        "year": year,
+    }
+
+
+def get_modern_archive_projects(year: int) -> tuple:
+    """
+    Retrieves project metadata for a year after 2016, if available
+    :param year: Year to retrieve project metadata from
+    :return: A tuple, with the first element a boolean which is True if we found project metadata for the year and
+    false otherwise (which it may be for the current year), and the second element an iterable of project metadata
+    """
+    org_url = f"https://summerofcode.withgoogle.com/api/archive/programs/{year}/organizations/"
+    orgs = requests.get(org_url).json()
+    is_ok = (type(orgs) == list) and (len(orgs) > 0)
+    if not is_ok:
+        return False, []
+    meta = (get_modern_archive_project(year, org["slug"]) for org in orgs)
+    return True, meta
+
+
+def get_curr_year_project(year: int, slug: str) -> dict:
+    """
+    Get project metadata for a project in the current year
+    :param year: The current year
+    :param slug: GSOC project slug
+    :return: Dict of project metadata
+    """
+    time.sleep(SCRAPE_DELAY)
+    project_url = f"https://summerofcode.withgoogle.com/api/organization/{slug}/"
+    meta = requests.get(project_url).json()
+    repos = []
+    repos.extend(get_link_matches(meta["description"]))
+    repos.extend(get_link_matches(meta["ideas_link"]))
+    repos.extend(get_link_matches(meta["source_code"]))
+    repos.extend(get_link_matches(meta["website_url"]))
```

Comment:
```suggestion
    for field in ["description", "ideas_link", "source_code", "website_url"]:
        repos.extend(get_link_matches(meta[field]))
```
Just a thought. Equally readable but less -> better. But I could be missing something :)